test/heterarchy.spec.coffee

```diff
@@ -58,42 +58,56 @@ describe 'heterarchy', ->
         constructor: ->
             @a = 'a'
         method: -> "A"
+        @method: -> "A"
+        @overrideNoSuper: () -> "a"
 
     class B extends A
         constructor: ->
             super
             @b = 'b'
         method: -> "B>#{super}"
+        @method: -> "B>#{super}"
+        @overrideNoSuper: () -> "b"
 
     class C extends A
         constructor: ->
             super
             @c = 'c'
         method: -> "C>#{super}"
+        @method: -> "C>#{super}"
+        @overrideNoSuper: () -> "c"
 
     class D extends multi B, C
         constructor: ->
             super
             @d = 'd'
         method: -> "D>#{super}"
+        @method: -> "D>#{super}"
+        @overrideNoSuper: () -> "d"
 
     class E extends A
         constructor: ->
             super
             @e = 'e'
         method: -> "E>#{super}"
+        @method: -> "E>#{super}"
+        @overrideNoSuper: () -> "e"
 
     class F extends multi C, E
         constructor: ->
             super
             @f = 'f'
         method: -> "F>#{super}"
+        @method: -> "F>#{super}"
+        @overrideNoSuper: () -> "f"
 
     class G extends multi D, F
         constructor: ->
             super
             @g = 'g'
         method: -> "G>#{super}"
+        @method: -> "G>#{super}"
```

Comment:
It would be nice to give the class method a different name, to make sure the methods checking instance methods are not passing because of the override.